CMakeLists.txt

```diff
@@ -119,7 +119,12 @@ if(CLANG_TIDY)
   #  append ";--fix" to arguments above to apply automatic fix (if the given checks support it)
 endif(CLANG_TIDY)
 
-add_executable(OpenSCAD)
+if(BUILD_SHARED_LIBS)
```

Comment:
Instead of one or the other, both should be made. It may help to look at #6299 because I got it to produce a static library and binary with minimal changes. Should be easy to make a shared library from the static one?

Reply:
Also, the value of this PR isn't to build a _shared_ library, but to build a library at all. Controlling whether to build a static libs vs. shared lib is what's usually controlled by the `BUILD_SHARED_LIBS` CMake variable.

Reply:
The submodules only use this to switch between static vs dynamic libraries. I think we should still have an option for building OpenSCAD as a static library, which is currently missing.

Reply:
Please rebase off #6299. The changes I have there is to build library targets `OpenSCADLibInternal` and `svg` which expose all APIs. You can create a new target `OpenSCADLib` (or whatever) which exposes the `lib_openscad` you are trying to publish.